### PR TITLE
8275648: Linker naming bikeshed

### PR DESCRIPTION
**Pull request by JornVernee**
Hi,

This PR renames several classes, functions, and files in the linker implementation. So far this has been avoided while the implementation was still being worked out. But now that we are moving to preview state, it is time to revisit these names.

Roughly speaking, the following name changes are applied:

  - ProgrammableInvoker           -> DowncallLinker
  - ProgrammableUpcallHandler -> UpcallLinker
  - 'native invoker'                      -> 'downcall stub'
  - 'optimzed upcall stub'            -> 'upcall stub'
  - OptimizedEntryBlob               -> UpcallStub
  - optimized_entry_frame           -> upcall_stub_frame

Then, some source files in hotspot are also renamed as follows:

  - universalNativeInvoker* -> downcallLinker*
  - universalUpcallHandler* -> upcallLinker*
  - foreign_globals*             -> foreignGlobals* (to match existing convention)
  
I've also fixed up some outdated comments.

Thanks,
Jorn

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8275648](https://bugs.openjdk.java.net/browse/JDK-8275648): Linker naming bikeshed

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/654/head:pull/654` \
`$ git checkout pull/654`

Update a local copy of the PR: \
`$ git checkout pull/654` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/654/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 654`

View PR using the GUI difftool: \
`$ git pr show -t 654`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/654.diff">https://git.openjdk.java.net/panama-foreign/pull/654.diff</a>

</details>

### CODE REVIEW

**Review by mcimadamore — COMMENTED** [review]
Looks good - one suspicious reference to PUH (see comment)

File: src/hotspot/share/prims/upcallLinker.cpp [hunk 1]
```diff
@@ -120,15 +120,15 @@ void ProgrammableUpcallHandler::on_exit(OptimizedEntryBlob::FrameData* context)
...
+void UpcallLinker::handle_uncaught_exception(oop exception) {
   ResourceMark rm;
   // Based on CATCH macro
   tty->print_cr("Uncaught exception:");
   exception->print();
   ShouldNotReachHere();
 }
 
-JVM_ENTRY(jlong, PUH_AllocateOptimizedUpcallStub(JNIEnv *env, jclass unused, jobject mh, jobject abi, jobject conv,
+JVM_ENTRY(jlong, PUH_MakeUpcallStub(JNIEnv *env, jclass unused, jobject mh, jobject abi, jobject conv,
```
[the remark this answers is not in the document]
Comment (JornVernee) [reply src/hotspot/share/prims/upcallLinker.cpp:131]: Right, good catch!

**Review by mcimadamore — APPROVED** [review]
All good on Linux.

**Pull request closed without merging**